paladin/mod.ts

```diff
@@ -0,0 +1,186 @@
+import { Drash } from "../deps.ts";
+
+type ReferrerPolicy =
+  "" |
+  "no-referrer" |
+  "no-referrer-when-downgrade" |
+  "same-origin" |
+  "origin" |
+  "strict-origin" |
+  "origin-when-cross-origin" |
+  "strict-origin-when-cross-origin" |
+  "unsafe-url"
+
+interface Configs {
+  "X-XSS-Protection"?: boolean
+  "Referrer-Policy"?: ReferrerPolicy,
+  "X-Content-Type-Options"?: boolean
+  hsts?: {
+    maxAge?: boolean | number,
+    includeSubDomains?: boolean,
+    preload?: boolean
+  },
+  "X-Powered-By"?: boolean | string
+  "X-Frame-Options"?: "DENY" | "SAMEORIGIN" | boolean | string, // eg ALLOW-FROM www.example.com
+  expectCt?: {
+    enforce?: boolean,
+    maxAge?: number,
+    reportUri?: string
+  },
+  "X-DNS-Prefetch-Control"?: boolean,
+  "Content-Security-Policy"?: string
+}
+
+/**
+ * A logger middleware inspired by https://www.npmjs.com/package/morgan.
+ *
+ * @param configs - See IDexterConfigs
+ */
+export function Paladin(
+  configs?: Configs,
+) {
+
+  if (!configs) {
+    configs = {}
+  }
+  if (!configs.hsts) {
+    configs.hsts = {}
+  }
+  if (!configs.expectCt) {
+    configs.expectCt = {}
+  }
+
+  // Default configs when no `configs` param is passed in
+  const defaultConfigs = {
+    "X-XSS-Protection": "1; mode=block",
+    "X-Content-Type-Options": "nosniff",
+    hsts: {
+      maxAge: 5184000, // 60 days
+      includeSubDomains: "includeSubDomains"
+    },
+    "X-Powered-By": false, // False for removing the header
+    "X-Frame-Options": "SAMEORIGIN",
+    "X-DNS-Prefetch-Control": false
+  };
+
+  /**
+   * The middleware function that's called by Drash.
+   *
+   * @param request - The request object.
+   * @param response - (optional) The response object.
+   */
+  function paladin(
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ): void {
+
+    // If there is no response, then we know this is occurring before the request
+    if (!response) {
```

Comment:
hahah can we just remove this block?

Reply:
hahaha no

Reply:
LOL alright then

Reply:
https://github.com/drashland/deno-drash-middleware/pull/25/commits/b8b3bba86166d57696b710bdb3513861ded928cd